### PR TITLE
15805 version in azure events

### PR DESCRIPTION
**Pull request by jack-h-wang**
This PR makes the current commit hash available to the main project and adds it to Azure events.

Test Steps:
1. Run Gradle task `package` or `quickPackage` and ensure the project builds successfully.

## Changes
- Add Gradle task to generate a Version object usable in the main project.
- Add version to `ReportEventData` and supply the commit hash from the Version object.

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] Added tests?

## Linked Issues
- #15805

## To be done
- #15989

### CODE REVIEW

File: prime-router/build.gradle.kts [hunk 1]
```diff
@@ -48,6 +48,16 @@ plugins {
...
+// retrieve the current commit hash
+val commitId by lazy {
+    val stdout = ByteArrayOutputStream()
+    exec {
+        commandLine("git", "rev-parse", "--short", "HEAD")
+        standardOutput = stdout
+    }
+    stdout.toString(StandardCharsets.UTF_8).trim()
+}
+
```
Comment (thetaurean) [inline prime-router/build.gradle.kts:60]: Does this end up running in CI? Are we sure that all the contexts this runs in has git?

As an alternative to running git, any thoughts on reading `.git/HEAD` and `.git/refs/<current-HEAD`?

File: prime-router/build.gradle.kts [hunk 2]
```diff
@@ -502,16 +516,29 @@ tasks.azureFunctionsPackage {
...
+        package gov.cdc.prime.router.version
+        
+        /**
+         * Supplies information for the current build.
+         * This file is generated via Gradle task prior to compile time and should always contain the current commit hash.
+         */
+        object Version {
+          const val commitId = "$commitId"
+        }
+    """.trimIndent()
```
Comment (thetaurean) [inline prime-router/build.gradle.kts:538]: Seems a tad sketchy to be writing out Kotlin here.

Any thoughts on making the code static and read from the version.json file that is output? Or perhaps read directly from `.git/HEAD` and use that info to get the hash from `.git/refs` ?

File: prime-router/src/test/kotlin/azure/ActionHistoryTests.kt [hunk 3]
```diff
@@ -352,7 +352,8 @@ class ActionHistoryTests {
             Topic.TEST,
             "http://blobUrl",
             TaskAction.send,
-            OffsetDateTime.now()
+            OffsetDateTime.now(),
+            ""
```
Comment (thetaurean) [inline prime-router/src/test/kotlin/azure/ActionHistoryTests.kt:356]: Any reason we shouldn't put at least a dummy version here? Ditto in other 4 locations.

**Review by thetaurean — COMMENTED** [review]
Just a couple minor questions / suggestions.

File: prime-router/build.gradle.kts (hunk 2 above)
Comment (jack-h-wang) [reply prime-router/build.gradle.kts:538]: This approach was suggested by @mkalish and is not dissimilar to the way jOOQ generates objects representing the database. I'm also not entirely certain the application can read files from those locations - the pivot to this approach was done amid difficulty attempting to implement exactly that.

File: prime-router/src/test/kotlin/azure/ActionHistoryTests.kt (hunk 3 above)
Comment (jack-h-wang) [reply prime-router/src/test/kotlin/azure/ActionHistoryTests.kt:356]: It didn't seem relevant to these specific tests; I only added references where it was needed.

File: prime-router/build.gradle.kts (hunk 1 above)
Comment (thetaurean) [reply prime-router/build.gradle.kts:60]: Nah if we know it works in all environments (which is obvious if already in master) then I'm good 👍.

**Review by thetaurean — APPROVED** [review]
👍

**Review by arnejduranovic — COMMENTED** [review]
Once deployed, let's verify that the commit in the events matches the commit in the json file by logging in to SSH (process described in deployment doc)

File: prime-router/build.gradle.kts [hunk 4]
```diff
@@ -502,16 +516,29 @@ tasks.azureFunctionsPackage {
 
 tasks.register("generateVersionFile") {
```
[the remark this answers is not in the document]
Comment (jack-h-wang) [reply prime-router/build.gradle.kts:517]: Added the TODO and created #15989.

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**